src/factory.ts

```diff
@@ -642,6 +656,17 @@ export class Factory {
     return system;
   }
 
+  /**
+   * Creates EasyScore. Normally the first step after constructing a factory.
+   *
+   * Example:
+   * `const vf: Factory = new Vex.Flow.Factory({renderer: { elementId: 'boo', width: 1200, height: 600 }});`
+   * `const score: EasyScore = vf.EasyScore();`
+   * @param params.factory not required
+   * @param params.builder TODO @0xfe
+   * @param params.commitHooks TODO @0xfe
```

Comment:
@0xfe what is the aim of these parameters?

Reply:
`EasyScore` uses a `Parser` to parse EasyScore `Grammar` and a `Builder` to actually build the VF elements.

A `commitHook` is like an extension, that's called to set additional properties on an element (usually a note) after it is built. E.g., `setId` sets the `id` of a note.

Reply:
The builder uses the `Factory` to build the elements, it doesn't build them directly. (This also keeps the Factory API consistent.)

Reply:
Thanks, what would you write in the comment then?

Reply:
No need to be too detailed here. This is sufficient:

builder: Instance of Builder
commitHooks: Function to call after a note element is created

Reply:
done